pkg/infrastructure/kogito_service.go

```diff
@@ -103,3 +105,20 @@ func getServiceEndpoints(client *client.Client, namespace string, serviceHTTPRou
 	}
 	return
 }
+
+// FetchKogitoServiceURI provide kogito service URI for given instance name
+func FetchKogitoServiceURI(cli *client.Client, name, namespace string) (string, error) {
+	log.Debugf("Fetching kogito service instance URI.")
+	service := &v1.Service{}
+	if exits, err := kubernetes.ResourceC(cli).FetchWithKey(types.NamespacedName{Name: name, Namespace: namespace}, service); err != nil {
+		return "", err
+	} else if !exits {
+		return "", fmt.Errorf("service with name %s not exist for Kogito service instance in given namespace %s", name, namespace)
+	}
+	port := service.Spec.Ports[0]
+	// resolves to http://dataindex.mynamespace.svc.cluster.local:8080 for example
+	uri := fmt.Sprintf("http://%s.%s.svc.cluster.local:%d", service.Name, service.Namespace, port.TargetPort.IntVal)
```

Comment:
During the weekend I made some tests around this and we can surpress the `svc.cluster.local` part. People can change this name in the kube configuration to differentiate between zones on hybrid installations, so I believe we need to remove it also.

Just this will work:

```suggestion
	uri := fmt.Sprintf("http://%s.%s:%d", service.Name, service.Namespace, port.Port)
```

Also `TargetPort` it's the port which the svc target in the container, not the exposed one. To use the exposed port we need to use `service.Port` instead. Soon we will use the default HTTP port and adding the port to the service won't be needed any more. See: https://issues.redhat.com/browse/KOGITO-3516

Reply:
Depends on: #598 to remove the port part.

Reply:
Updated to suggested change